File: problems/1165/solution.go
package main
func calculateTime(keyboard string, word string) int {
    m, i, ans := map[rune]int{}, 0, 0
    
    for i, ch := range keyboard {
        m[ch] = i
    }
    
    for _, ch := range word {
        j := m[ch]
        ans, i = ans + abs(i - j), j
        
    }
    
    return ans
}

func abs(a int) int {
    if a < 0 { return a * -1 }
    return a
}

File: problems/1165/solution.cpp
class Solution {
public:
    int calculateTime(string keyboard, string word) {
        int ans = 0, lastPos = 0;
        vector<int> positions(26);
        
        for (int i = 0; i < keyboard.size(); i++) {
            positions[keyboard[i] - 'a'] = i;
        }
        
        for (const char& ch: word) {
            ans += abs(lastPos - positions[ch - 'a']);
            lastPos = positions[ch - 'a'];
        }
        
        return ans;
    }
};